amethyst_core/src/transform/components/local_transform.rs

```diff
@@ -280,158 +353,166 @@ impl Component for Transform {
 impl From<Vector3<f32>> for Transform {
     fn from(translation: Vector3<f32>) -> Self {
         Transform {
-            translation,
+            iso: Isometry3::new(translation, na::zero()),
             ..Default::default()
         }
     }
 }
 
-impl CgTransform<Point3<f32>> for Transform {
-    fn one() -> Self {
-        Default::default()
-    }
-
-    fn look_at(eye: Point3<f32>, center: Point3<f32>, up: Vector3<f32>) -> Self {
-        let rotation = Quaternion::look_at(center - eye, up);
-        let translation = rotation.rotate_vector(Point3::origin() - eye);
-        let scale = Vector3::from_value(1.);
-        Self {
-            scale,
-            rotation,
-            translation,
-        }
-    }
-
-    fn transform_vector(&self, vec: Vector3<f32>) -> Vector3<f32> {
-        self.rotation
-            .rotate_vector(vec.mul_element_wise(self.scale))
-    }
-
-    fn inverse_transform_vector(&self, vec: Vector3<f32>) -> Option<Vector3<f32>> {
-        if ulps_eq!(self.scale, &Vector3::zero()) {
-            None
-        } else {
-            Some(
-                self.rotation
-                    .invert()
-                    .rotate_vector(vec.div_element_wise(self.scale)),
-            )
-        }
-    }
-
-    fn transform_point(&self, point: Point3<f32>) -> Point3<f32> {
-        let p = Point3::from_vec(point.to_vec().mul_element_wise(self.scale));
-        self.rotation.rotate_point(p) + self.translation
-    }
-
-    fn concat(&self, other: &Self) -> Self {
-        Self {
-            scale: self.scale.mul_element_wise(other.scale),
-            rotation: self.rotation * other.rotation,
-            translation: self
-                .rotation
-                .rotate_vector(other.translation.mul_element_wise(self.scale))
-                + self.translation,
+impl<'de> Deserialize<'de> for Transform {
```

Comment:
Nice job on this!

Reply:
Is that ironic or do you actually mean it? @torkleyy complained about it so I'm not too sure :sweat_smile: